lib/class-wp-theme-json-resolver-gutenberg.php

```diff
@@ -760,4 +762,34 @@ public static function get_style_variations() {
 		}
 		return $variations;
 	}
+
+	/**
+	 * Resolves relative paths in theme.json styles to theme absolute paths.
+	 *
+	 * @since 6.6.0
+	 *
+	 * @param WP_Theme_JSON_Gutenberg  $theme_json A theme json instance.
+	 * @return WP_Theme_JSON_Gutenberg A theme json instance with resolved paths.
+	 */
+	protected static function resolve_theme_file_uris( $theme_json ) {
+		if ( empty( $theme_json ) ) {
+			return;
+		}
+
+		/*
+		 * Styles backgrounds.
+		 * Where a URL is not absolute (has no host fragment), it is assumed to be relative to the theme directory.
+		 * Blocks, elements, and block variations are not yet supported.
+		 */
+		if (
+			isset( $theme_json['styles']['background']['backgroundImage']['url'] ) &&
+			is_string( $theme_json['styles']['background']['backgroundImage']['url'] ) &&
+			! isset( wp_parse_url( $theme_json['styles']['background']['backgroundImage']['url'] )['host'] ) ) {
```

Comment:
Inspired by @draganescu's comments in https://github.com/WordPress/gutenberg/pull/60578#issuecomment-2084508953

Thank you!